app/javascript/orangelight/availability_show.js

```diff
@@ -0,0 +1,280 @@
+      this.request_scsb_single_availability();
+    } else {
+      this.request_show_availability(allowRetry);
+    }
+  }
+
+  request_scsb_single_availability() {
+    const url = `${this.availability_url}?scsb_id=${this.id.replace(/^SCSB-/, '')}`;
+    fetch(url)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error(`HTTP status: ${response.status}`);
+        }
+        return response.json();
+      })
+      .then((data) => {
+        this.process_scsb_single(data);
+      })
+      .catch((error) => {
+        console.error(
+          `Failed to retrieve availability data for the SCSB record ${this.id}: ${error}`
+        );
+      });
+  }
+
+  request_show_availability(allowRetry) {
+    const url = this.availability_url_show();
+    fetch(url)
+      .then((response) => {
+        if (response.status === 429) {
+          if (allowRetry) {
+            console.log(`Retrying availability for record ${this.id}`);
+            window.setTimeout(() => {
+              this.update_availability_retrying();
+              this.request_availability(false);
+            }, 1500);
+            return;
+          } else {
+            console.error(
+              `Failed to retrieve availability data for the bib (retry). Record ${this.id}: HTTP status 429`
+            );
+            this.update_availability_undetermined();
+            return;
+          }
+        }
+        if (!response.ok) {
+          throw new Error(`HTTP status: ${response.status}`);
+        }
+        return response.json();
+      })
+      .then((data) => {
+        if (data) {
+          this.process_single(data);
+        }
+      })
+      .catch((error) => {
+        console.error(
+          `Failed to retrieve availability data for the bib. record ${this.id}: ${error.message}`
+        );
+      });
+  }
+
+  /* example with three host ids: https://bibdata.princeton.edu/bibliographic/availability.json?deep=true&bib_ids=9923427953506421,99125038613506421,99125026373506421,99124945733506421 */
+  // the record id is 9923427953506421
+  availability_url_show() {
+    let url = `${this.bibdata_base_url}/bibliographic/availability.json?deep=true&bib_ids=${this.id}`;
+    if (this.host_id.length > 0) {
+      const hostIds = Array.isArray(this.host_id)
+        ? this.host_id.join(',')
+        : this.host_id;
+      url += `,${hostIds}`;
+    }
+    return url;
+  }
+
+  // process_single() is used in the Show page and typically `holding_records` only has the
+  // information for a single bib since we are on the Show page. But occasionally the record
+  // that we are showing is bound with another (host) record and in those instances
+  // `holding_records` has data for two or more bibs: `this.id`, `this.host_id`.
+  process_single(holding_records) {
+    this.update_single(holding_records, this.id);
+    // Availability response in bibdata should be refactored not to include the host holdings under the mms_id of the record page.
```

Comment:
I think this is not true anymore. review the availability request and the connected ticket.